src/screens/Settings/FindContactsSettings.tsx

```diff
@@ -0,0 +1,562 @@
+          />
+          {!shadow.viewer?.following && (
+            <Button
+              color="secondary"
+              variant="ghost"
+              label={_(msg`Remove suggestion`)}
+              onPress={() => dismissMatch(profile.did)}
+              hoverStyle={[a.bg_transparent, {opacity: 0.5}]}
+              hitSlop={8}>
+              <ButtonIcon icon={XIcon} />
+            </Button>
+          )}
+        </ProfileCard.Header>
+      </View>
+    </View>
+  )
+}
+
+function StatusHeader({
+  numMatches,
+  isPending,
+  isAnyUnfollowed,
+}: {
+  numMatches: number
+  isPending: boolean
+  isAnyUnfollowed: boolean
+}) {
+  const {_} = useLingui()
+  const agent = useAgent()
+  const queryClient = useQueryClient()
+  const {currentAccount} = useSession()
+
+  const {
+    mutate: onFollowAll,
+    isPending: isFollowingAll,
+    isSuccess: hasFollowedAll,
+  } = useMutation({
+    mutationFn: async () => {
+      const didsToFollow = []
+
+      let cursor: string | undefined
+      do {
+        const page = await agent.app.bsky.contact.getMatches({
+          limit: 100,
+          cursor,
+        })
+        cursor = page.data.cursor
+        for (const profile of page.data.matches) {
+          if (
+            profile.did !== currentAccount?.did &&
+            !isBlockedOrBlocking(profile) &&
+            !isMuted(profile) &&
+            !profile.viewer?.following
+          ) {
+            didsToFollow.push(profile.did)
+          }
+        }
+      } while (cursor)
+
+      logger.metric('contacts:settings:followAll', {
+        followCount: didsToFollow.length,
+      })
+
+      const uris = await wait(500, bulkWriteFollows(agent, didsToFollow))
+
+      for (const did of didsToFollow) {
+        const uri = uris.get(did)
+        updateProfileShadow(queryClient, did, {
+          followingUri: uri,
+        })
+      }
+    },
+    onSuccess: () => {
+      Toast.show(_(msg`Followed all matches`), {type: 'success'})
+    },
+    onError: err => {
+      if (isNetworkError(err)) {
+        Toast.show(
+          _(
+            msg`Could not follow all matches - please check your network connection.`,
+          ),
+          {type: 'error'},
+        )
+      } else {
+        logger.error('Failed to follow all matches', {safeMessage: err})
+        Toast.show(_(msg`Could not follow all matches. ${cleanError(err)}`), {
+          type: 'error',
+        })
+      }
+    },
+  })
+
+  if (numMatches > 0) {
+    if (isPending) {
+      return (
+        <View style={[a.w_full, a.py_3xl, a.align_center]}>
+          <Loader size="xl" />
+        </View>
+      )
+    }
+
+    return (
+      <View
+        style={[
+          a.pt_xl,
+          a.px_xl,
+          a.pb_md,
+          a.flex_row,
+          a.justify_between,
+          a.align_center,
+        ]}>
+        <Text style={[a.text_md, a.font_semi_bold]}>
+          <Plural
+            value={numMatches}
+            one="1 contact found"
```

Comment:
```suggestion
            one="# contact found"
```
I think it might be better to have the `#` rather than `1`, because IIUC in some languages – including Russian, Scottish Gaelic and Ukrainian that we have localizations for – the `one` plural form is also used for other numbers than just 1 itself:

https://www.unicode.org/cldr/charts/48/supplemental/language_plural_rules.html